lib/read-from-url.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const tmp = require('tmp');
+const fetch = require('minipass-fetch');
+const fs = require('fs');
+const debug = require('debug')('supported:read-from-url');
+
+const RAW_CONTENT_HOST = 'raw.githubusercontent.com';
+
+async function setupProjectPath(_url, rawFileHost) {
+  let tempDir = tmp.dirSync();
+
+  let url = new URL(_url);
+  url.host = rawFileHost || RAW_CONTENT_HOST;
+  if (url.pathname.indexOf('tree') == -1) {
```

Comment:
Should we do a preflight check, to make sure this is a server that we support? That way we can give a good error message if we do not.

Reply:
Handled
```
$ bin/supported https://test.githubprivate.com/stefanpenner/supported -t='ada'
✨ working
Error: Couldn't reach server, please check the URL provided.
    request to https://ada@raw.test.githubprivate.com/stefanpenner/supported/master/package.json failed, reason: getaddrinfo ENOTFOUND raw.test.githubprivate.com
 ```